lib/helpers/schema/getIndexes.js

```diff
@@ -108,9 +110,11 @@ module.exports = function getIndexes(schema) {
       fixSubIndexPaths(schema, prefix);
     } else {
       schema._indexes.forEach(function(index) {
-        if (!('background' in index[1])) {
-          index[1].background = true;
+        const options = index[1];
+        if (!('background' in options)) {
+          options.background = true;
         }
+        decorateDiscriminatorIndexOptions(schema, options);
```

Comment:
Is it necessary to add the discriminator index options here as opposed to in `syncIndexes()`? As per our discussion in Slack, I'd like to avoid tying discriminators to schemas any more than we already do.